entities/withdraw_transaction.rb

```diff
@@ -0,0 +1,20 @@
+class WithdrawTransation < BaseTransaction
+  def run
+    @card.update_balance(new_balance) if valid?
+    WithdrawTransationResult.new(card: @card, amount: @amount, errors: errors)
+  end
+
+  private
+
+  def validate
+    if !@amount.positive?
```

Comment:
I think in case of zero it has to be valid. Don't it?
```suggestion
    if @amount.negative?
```

Reply:
In real life it is impossible to withdraw $0